ethkey/main.cpp

```diff
@@ -82,23 +86,48 @@ int main(int argc, char** argv)
 	setDefaultOrCLocale();
 	KeyCLI m(KeyCLI::OperationMode::ListBare);
 	g_logVerbosity = 0;
-
+	po::options_description generalOptions("General Options");
+	generalOptions.add_options()
+		("verbosity,v", po::value<string>(), "<0 - 9>  Set the log verbosity from 0 to 9 (default: 8).")
+		("version,V", "Show the version and exit.")
+		("help,h",  "Show this help message and exit.")
+	;
+	int ac = 1;
 	for (int i = 1; i < argc; ++i)
 	{
 		string arg = argv[i];
 		if (m.interpretOption(i, argc, argv)) {}
-		else if ((arg == "-v" || arg == "--verbosity") && i + 1 < argc)
-			g_logVerbosity = atoi(argv[++i]);
-		else if (arg == "-h" || arg == "--help")
-			help();
-		else if (arg == "-V" || arg == "--version")
-			version();
+		else if (((arg == "-v" || arg == "--verbosity") && i + 1 < argc) || arg == "-h" || arg == "--help" || arg == "-V" || arg == "--version") {
+			argv[ac] = argv[i];
```

Comment:
Could you explain what are you trying to achieve in this part?
I don't think assiging to `argv` is a very good idea

I think you're trying to get rid of all the options that are handled in `KeyCLI::interpretOption()` and leave only `-v`, `-V`, `-h` in `argv`
Why do you need this? Wouldn't `po::store(po::parse_command_line(ac, argv, generalOptions), vm);` below just ignore options not listed in `generalOptions` ?

Reply:
Also look at this http://www.boost.org/doc/libs/1_65_1/doc/html/program_options/howto.html#idp437834784